content/guides/azure-todo-static-web-app.md

```diff
@@ -0,0 +1,747 @@
+   - Handles todo item styling based on completion status
+   - Creates interactive elements (checkboxes, delete buttons)
+
+Utility Functions:
+
+1. `showLoading()`
+
+   - Displays loading indicator during API operations
+   - Provides visual feedback for better user experience
+
+2. `hideLoading()`
+
+   - Removes loading indicator after operations complete
+   - Prepares UI for content display
+
+3. `showError(message)`
+   - Displays error messages to users
+   - Implements auto-dismissing notifications (3-second timeout)
+   - Provides clear feedback for error conditions
+
+State Management:
+
+- All todo items are stored in the `todos` array
+
+Error Handling:
+
+- Comprehensive try-catch blocks around API operations
+- Detailed error messages for debugging
+
+### Testing the Frontend locally
+
+To test the frontend, in your terminal, run the following command to start a local server:
+
+```bash
+swa start
+```
+
+This will start a local server on [`http://localhost:4280`](http://localhost:4280) where you can access the todo app. Open the URL in your browser to be greeted with the initial todo list interface:
+
+![Todo App Initial State](/docs/guides/swa-todo-app-intial.png)
+
+\*Note that full functionality will be available once we set up the Azure Functions backend API.
+
+### Creating Azure Functions
+
+Azure Functions lets you write backend code that runs without managing any servers. We'll use it to create API endpoints that handle all our todo operations – creating, reading, updating, and deleting todos from our database. Think of these functions as small pieces of code that wake up when needed and automatically handle user requests.
+
+Create a new directory named `api` in the root of your project:
+
+```bash
+mkdir api && cd api
+```
+
+Press F1 in VS Code to open the command palette and run the command `Azure Static Web Apps: Create HTTP Function`. Choose `JavaScript` as the language with the Model version `v4`. Name the function `todos` when prompted.
+
+This will create a new Azure Function in the `api` directory, which will serve as the backend API for our todo application with an api endpoint `/api/todos`.
+
+We will be using the [`@neondatabase/serverless`](https://www.npmjs.com/package/@neondatabase/serverless) package to connect to the Neon Postgres database. Install the package by running the following command in the `api` directory:
+
+```bash
+npm install @neondatabase/serverless
+```
+
+Edit the `api/src/functions/todos.js` file to add the following code:
+
+```javascript
+const { app } = require('@azure/functions');
+const { neon } = require('@neondatabase/serverless');
+
+const sql = neon(process.env.DATABASE_URL);
+
+// Helper function to get todos and create table if it doesn't exist
+const getTodos = async () => {
+  try {
+    const todos = await sql`SELECT * FROM todos`;
+    return todos;
+  } catch (error) {
+    if (error.code === '42P01') {
+      // Table does not exist, so create it
+      await sql`
```

Comment:
<img width="414" alt="Screenshot 2024-12-17 at 12 34 45 AM" src="https://github.com/user-attachments/assets/924bc556-16ce-485e-a21c-ed85d6f137e0" />

may we fix the formatting of this?

Reply:
I hope its fine now